executor/src/witgen/jit/quadratic_symbolic_expression.rs

```diff
@@ -411,58 +418,54 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display> QuadraticSymbolicExp
         Some(Effect::RangeConstraint(solve_for.clone(), constraint))
     }
 
-    /// Tries to detect a bit constraint of the form `X * (X - 1) = 0`.
+    /// Tries to detect a range constraint of the form `(X - value1) * (X - value2) = 0`.
+    /// The most common example would be a bit constraint.
     /// In that case, we can return a range constraint for `X` and mark the
     /// constraint as complete.
-    // TODO: This function would not detect many equivalent constraints, like
-    // `X * X - X` or `5 * X * (3 * X - 3) = 0`.
+    // TODO: This function would not detect some equivalent constraints, like
+    // `X * X - X`.
     // This could be fixed by finding a canonical form for the quadratic
     // expression, and normalizing the constants.
-    fn try_detect_bit_constraint(&self) -> Option<ProcessResult<T, V>> {
-        // Bit range constraints can be implemented as `X * (X - 1) = 0`.
+    fn try_detect_range_constraint(&self) -> Option<ProcessResult<T, V>> {
         if !self.linear.is_empty() || !self.constant.is_known_zero() || self.quadratic.len() != 1 {
             return None;
         }
 
-        // Looking for `X * (X - 1)`
         let (factor1, factor2) = &self.quadratic[0];
```

Comment:
Could use https://github.com/powdr-labs/powdr/pull/2643/files#diff-3a618405981b7333007d0edb2a8fa1c5302b3a4b04243ba09bde80a213182e94R123 (not merged yet)